src/tools/lifter/models/l3mod/m0_mod_stepScript.sml

```diff
@@ -0,0 +1,110 @@
+open HolKernel Parse boolLib bossLib;
+
+open arithmeticTheory;
+open wordsTheory;
+open combinTheory;
+
+(* This theory contains the definition of the modified m0 step
+   (, where clock cycles are counted with word64 instead if num). *)
+
+val _ = new_theory "m0_mod_step";
+
+
+(* modified state with word representation of clock cyckle counter *)
+val _ = Datatype `m0_mod_state = <|
+  base   : m0_state;
+  countw : word64
+|>`;
+
+
+(* Definitions *)
+val m0_mod_def = Define `
+  m0_mod s = if s.count < (2 ** 64) then SOME (<| base := s; countw := n2w s.count |>) else NONE
+`;
+
+val m0_mod_inv_def = Define `
+  m0_mod_inv sm = sm.base with <| count := w2n sm.countw |>
+`;
+
+val NextStateM0_mod_def = Define `
+                     NextStateM0_mod sm = case NextStateM0 (m0_mod_inv sm) of
```

Comment:
All of a sudden, the `Define`ition is indented :thinking: 
:stuck_out_tongue_closed_eyes: